plugins/database/mysql/connection_producer.go

```diff
@@ -29,6 +35,8 @@ type mySQLConnectionProducer struct {
 	MaxConnectionLifetimeRaw interface{} `json:"max_connection_lifetime" mapstructure:"max_connection_lifetime" structs:"max_connection_lifetime"`
 	Username                 string      `json:"username" mapstructure:"username" structs:"username"`
 	Password                 string      `json:"password" mapstructure:"password" structs:"password"`
+	AuthType                 string      `json:"auth_type" mapstructure:"auth_type" structs:"auth_type"`
+	ServiceAccountJSON       string      `json:"service_account_json" mapstructure:"service_account_json" structs:"service_account_json"`
```

Comment:
nit: I was going to suggest adding a comment to this field to document that it is only for GCP. But maybe we should consider renaming it instead? I think in Vault-land we usually use `GOOGLE_APPLICATION_CREDENTIALS` for service account JSON.

Reply:
Interesting point! This field was originally named `credentials`, and was renamed to `service_account_json` with the same intent of making it GCP specific. I would say `service_account_json` suffices enough to make it explicit that this is a GCP related field because in the IAM context, I think only GCP uses the term 'service accounts' as far as I know 🤔 (I believe AWS just calls them IAM users). If it is not clear enough I can add in a comment like you suggested!

Here's the original renaming discussion for context: https://github.com/hashicorp/vault/pull/22445#discussion_r1311962192

Reply:
We're still making use of the `GOOGLE_APPLICATION_CREDENTIALS` under the hood as the environment variable though, since that is what the `gcloud` sdk also expects (similar to Vault)